src/librustc_resolve/lib.rs

```diff
@@ -3457,6 +3521,98 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         found_traits
     }
 
+    // Looks up a definitions (and paths) based on a user-supplied predicate
+    // This method should be used to generate suggestions when name resolution
+    // fails.
+    fn lookup_suggestions<FilterFn>(&mut self,
+                                    lookup_name: Name,
+                                    lookup_type: LookupTarget,
+                                    namespace: Namespace,
+                                    filter_fn: FilterFn) -> Vec<Path>
+        where FilterFn: Fn(Def) -> bool {
+
+        let mut lookup_results: Vec<Path> = Vec::new();
+        let mut lookup_modules: Vec<(Module, Vec<PathSegment>)> = Vec::new();
```

Comment:
Note that these type annotations aren't necessary (though they don't hurt, always nice when reading to know the types of your vectors).